edb/pgsql/compiler/relgen.py

```diff
@@ -1389,6 +1392,11 @@ def process_set_as_membership_expr(
         left_is_row_expr = astutils.is_row_expr(left_out)
 
         with newctx.subrel() as _, _.newscope() as subctx:
+            if is_array_unpack:
+                relctx.update_scope(orig_right_arg, subctx.rel, ctx=subctx)
```

Comment:
Why do we need to `update_scope` on both `orig_` and `unwrapped_`?

Reply:
Because if `unwrapped_` has a `path_scope_id`, then that takes precedence, but otherwise we want to use `orig_`'s